pkg/generator/certificate.go

```diff
@@ -37,28 +49,85 @@ func generateCA(alias *v1alpha1.AliasConfig) ([]byte, error) {
 	}
 	privateKey, err := ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
 	if err != nil {
-		return []byte{}, err
+		return RootCA{}, err
 	}
 
-	ca, err := x509.CreateCertificate(rand.Reader, template, template, &privateKey.PublicKey, privateKey)
+	ca, err := x509.CreateCertificate(rand.Reader, caTemplate, caTemplate, &privateKey.PublicKey, privateKey)
 	if err != nil {
-		return ca, err
+		return RootCA{}, err
 	}
-	caPEM := new(bytes.Buffer)
-	pem.Encode(caPEM, &pem.Block{
+	caPEM := pem.EncodeToMemory(&pem.Block{
 		Type:  "CERTIFICATE",
-		Bytes: ca,
-	})
-	// TODO do we need caPrivateKeyPEM?
+		Bytes: ca})
+
+	marshaledPrivateKey, err := x509.MarshalECPrivateKey(privateKey)
+	if err != nil {
+		return RootCA{}, err
+	}
+	caPrivateKeyPEM := pem.EncodeToMemory(&pem.Block{
+		Type:  "EC PRIVATE KEY",
+		Bytes: marshaledPrivateKey})
+
+	//TODO: Which way do we want to encode the PEM for the ca?
 	// marshaledPrivateKey, err := x509.MarshalPKCS8PrivateKey(privateKey)
 	// if err != nil {
-	//     return ca, err
+	// 	return caPEM, []byte{}, err
 	// }
-	// caPrivateKeyPEM := new(bytes.Buffer)
-	// pem.Encode(caPrivateKeyPEM, &pem.Block{
-	//     Type:  "RSA PRIVATE KEY",
-	//     Bytes: marshaledPrivateKey,
-	// })
+	// caPrivateKeyPEM := pem.EncodeToMemory(&pem.Block{
+	// 	Type:  "RSA PRIVATE KEY",
+	// 	Bytes: marshaledPrivateKey})
+
+	return RootCA{CAPem: caPEM, CAPrivateKeyPEM: caPrivateKeyPEM, CA: caTemplate, CAKey: privateKey}, nil
```

Comment:
JFYI, you can also do this if you prefer, whatever you think it more readable. I could go either way.
```suggestion
	return RootCA{caPEM, caPrivateKeyPEM, caTemplate, privateKey}, nil
```